test/SimulationCSharpClient.Tests/HttpRetryHandler/HttpRetryMessageHandlerTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Moq;
+using Moq.Protected;
+using NUnit.Framework;
+using SimulationCSharpClient.Client;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SimulationCSharpClient.Tests.HttpRetryHandler
+{
+    [TestFixture]
+    public class HttpRetryMessageHandlerTests
+    {
+        [Test]
+        public void HttpGet_Success()
+        {
+            // Arrange
+            var retryMessageHandler = new HttpRetryMessageHandler(6);
+
+            using (var httpClient = new HttpClient(retryMessageHandler))
+            {
+                // Act
+                var response = httpClient.GetStringAsync("http://www.google.com").Result;
```

Comment:
should not rely upon an internet connection in general for unit tests. would be better to mock the httpclient

Reply:
Resolved by creating TestHandlers for Success and Failures and with Dummy Apis